# SingleRMEA: Elucidating Gene Expression of Cells from Single-Cell RNA-seq data

The development of single-cell RNA-seq technologies provides new opportunities for biology since it has become an accepted experimental method throughput improvements enabling applications for cell type discovery. However,  high-throughput applications of single-cell RNA-seq to solid tissues rely on the formal cell type definitions. Unfortunately, it is unclear how to formulate such definitions to analyze the crucial information on cells' location further since high levels of technical noise in most data. To address this challenge, we present a computational method (SingleRMEA) to conduct the robust cell type classifiers including clustering and classification to address large-scale PBMC dataset and human tissue sources composed of complex mixtures of cell types and subtypes. For clustering, clustering by fast search and find of density peaks (CDP) is employed to perform clustering to partition the cells into a few distinct subpopulations. For classification, we develop a new ensemble construction method to predict the cell type for single-cell RNA-seq, which applies multiobjective optimization to the stacking ensemble construction process to generate domain-specific configurations under two objective functions. To validate our SingleRMEA method, we compare its performance across two PBMCs datasets including PBMC-4k and PBMC-12 merged data. The experimental results demonstrate that the SingleRMEA can obtain superior performance over the current state-of-the-art methods. Meanwhile, it also demonstrates that SingleRMEA can enable the construction of cell type classifiers that can be direct to other new single-cell RNA-seq data. SingleRMEA is written in Matlab and available at https://github.com/lixt314/SingleRMEA.

Data
We have collected the immune cell single-cell RNA-seq profiles across the human datasets from \citet{schelker2017estimation}. In particular, two PBMC-merged data sources are considered by interpreting three different PBMCs data with the melanoma patient samples and ovarian cancer ascites samples. The PBMCs data includes PBMC-4k and PBMC-12k. The first dataset (PBMC-4k merged) includes 4k PBMCs \footnote{https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc4k} from a healthy donor, which contains 4000 single cells sequenced on Illumina Hiseq4000 with approximately 87,000 reads per cell \cite{zheng2017massively}, 4645 tumor-derived single cells from 19 melanoma patient samples, 3114 single cells from four ovarian cancer ascites samples. The second dataset (PBMC-12k merged) contains the 4k PBMCs and the 8k PBMCs\footnote{https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc8k} from a healthy donor, melanoma patient samples, and four ovarian cancer ascites samples. The 8k PBMCs has 8,381 cells detected sequenced on Illumina Hiseq4000 with approximately 92,000 reads per cell \cite{zheng2017massively}.  The performance measure reported by 10-fold cross-validation is the average of the values calculated.
